source/Halibut/DataStream.cs

```diff
@@ -149,12 +152,14 @@ public static DataStream FromStream(Stream source, Action<int> updateProgress)
             });
         }
         
+        [Obsolete]
         public static DataStream FromStream(Stream source, Action<int> updateProgress, Func<int, CancellationToken, Task> updateProgressAsync)
```

Comment:
Should this be marked as obsolete? The caller gives it everything that is required and halibut will call the sync or async methods based on what its doing. We might want to mark this as obsolete at the time sync is removed from halibut.

This method would be used now in Octopus since it would give us a better feature toggle, as we can keep the sync implementation of `updateProgress`.

Reply:
> We might want to mark this as obsolete at the time sync is removed from halibut.

This may be a little late, as it would be deleted!! 

It's to ensure that we are aware of all callers to this operation, as it does have sync bits.

Reply:
I was thinking the existing code would be:
```
DataStream.FromStream(source, ExistingSyncProgressMethod);
```
Then with this method we could have:
```
DataStream.FromStream(source, ExistingSyncProgressMethod, NewAsyncProgressMessage);
```
So `NewAsyncProgressMessage` will only be called if async is enabled. If sync is enabled then the existing `ExistingSyncProgressMethod` would be called. That is the toggle.

If we don't use this method now then we would end up with:
```
DataStream.FromStream(source, NewAsyncProgressMessage);
```
and the toggle would not let us revert to `ExistingSyncProgressMethod`

Reply:
Ok, I'll remove this obsolete